hazelcast/src/main/java/com/hazelcast/internal/partition/MigrationEventHandler.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition;
+
+import com.hazelcast.partition.MigrationListener;
+import com.hazelcast.partition.MigrationState;
+import com.hazelcast.partition.ReplicaMigrationEvent;
+
+public final class MigrationEventHandler {
+
+    public static final int MIGRATION_STARTED_PARTITION_ID = -1;
+    public static final int MIGRATION_FINISHED_PARTITION_ID = -2;
+
+    private final MigrationListener migrationListener;
+
+    public MigrationEventHandler(MigrationListener migrationListener) {
+        this.migrationListener = migrationListener;
+    }
+
+    public void handleMigrationEvent(MigrationState state, MigrationEventType type) {
+        switch (type) {
+            case MIGRATION_STARTED:
+                migrationListener.migrationStarted(state);
+                break;
+            case MIGRATION_FINISHED:
+                migrationListener.migrationFinished(state);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid event type: " + type);
+        }
+    }
+
+    public void handleReplicaMigrationEvent(ReplicaMigrationEvent event) {
+        switch (event.getPartitionId()) {
```

Comment:
Hi @peterjot 
I have just realized the reason for using the partition id here. I did not know that we are using negative values of partition id as the event type. I have internally opened up this issue with the team. And we decided that in the client protocol we want to separate the event type from the partition id. So we are on the right track. 

If I am not mistaken, we should not need to check the partition id anymore in this path, since we already made sure on the server we will not see Migration_started -1 and migration finished -2 on the partition id's. 

When you remove that this class will be fairly small. Can you remove this class and move the logic to  
`PartitionServiceProxy`. I think we don't need a new class and redirection for a couple of lines anymore.